### DISCUSSION TITLE
Ability to stitch static data to SSR pages

### DISCUSSION THREAD

**Discussion by collegewap** [vercel/next.js#15095] [Ideas]
**Problem:** Currently there is no way to have a particular section of a page to be statically rendered and remaining part of the page to be rendered dynamically using SSR

**Usecase:** In the listing page of an e-commerce application, I would want the products (one inside the red box, apologies for the horrible diagram 🙂) to be rendered through SSR (cannot render through SSG because, if done through SSG pagination needs to be CSR and it will impact SEO). And I would like to have the category flyout (one inside the green box) to be statically generated during build time and kept in the file system and accessed during SSR.

![image](https://user-images.githubusercontent.com/9767383/87242964-3b9fa480-c44f-11ea-8e18-fbac4fb20622.png)

This would avoid querying and curating category hierarchy data on every page load. This will help not just the listing page, but all the SSR pages where the flyout is displayed

**Possible Solution**: 

- Can have a JSON file generated during build time, which is accessed during SSR

- Can have am HTML file created for the section during build time and rehydrated during SSR.

I went through the discussion https://github.com/vercel/next.js/discussions/11424, and it is suggested to use the middleware feature (https://github.com/vercel/next.js/issues/7208). Looks like it was reverted later (https://github.com/vercel/next.js/pull/12218).

I am aware of the workarounds like 

- Having a webhook to create/update the JSON file in the codebase and trigger a build whenever the category data changes.
- Fetching the category data in the client-side to speed up the SSR since the flyout is not displayed until the user clicks the menu and user is highly unlikely to click before the CSR happens.

Would appreciate feedback!

**Comment by trompx** [comment]
Hey @collegewap, do you mind sharing the solution you ended up with?
